src/config.rs

```diff
@@ -225,6 +232,28 @@ impl Config {
                 .with_context(|| "Missing private key")
         }?;
 
+        let endpoint_addr = parse_addr(matches.value_of("endpoint-addr"))
+            .with_context(|| "Invalid endpoint address")?;
+
+        let endpoint_bind_addr = if let Some(addr) = matches.value_of("endpoint-bind-addr") {
+            let addr = parse_addr(Some(addr)).with_context(|| "Invalid host address")?;
+            // Make sure the host address and endpoint address are the same IP version
+            if addr.ip().is_ipv6() && endpoint_addr.ip().is_ipv6()
+                || (addr.ip().is_ipv4() && endpoint_addr.ip().is_ipv4())
```

Comment:
The logic here is inverted. Simply,

```suggestion
            if addr.ip().is_ipv4() != endpoint_addr.ip().is_ipv4()
```